test/splitter.js

```diff
@@ -0,0 +1,160 @@
+const truffleAssert = require('truffle-assertions');
+const Splitter = artifacts.require("./Splitter.sol");
+
+contract('Splitter', async accounts => {
+
+    let gasCost = async receipt => {
+        const gasUsed = receipt.receipt.gasUsed;
+        const tx = await web3.eth.getTransaction(receipt.tx);
+        const gasPrice = tx.gasPrice;
+
+        return web3.utils.toBN(gasUsed * gasPrice);
+    };
+
+    let checkEventNotEmitted = async eventName => {
+        let result = await truffleAssert.createTransactionResult(splitter, splitter.transactionHash);
+
+        await truffleAssert.eventNotEmitted(
+            result,
+            eventName
+        );
+    }
+
+    const senderAddress = accounts[1];
+    const RecipientOneAddress = accounts[2];
+    const RecipientTwoAddress = accounts[3];
+    let splitter;
+
+    beforeEach("deploy and prepare", async function() {
+        splitter = await Splitter.new();
+    });
+
+    it('Sender sends 5 wei, 4 wei is split equally between recipients and 1 wei sent to the senders balance', async () => {
+
+        // Take snapshot of initial balances
+        let initSenderEthBalance = await web3.eth.getBalance(senderAddress);
+        initSenderEthBalance = web3.utils.toBN(initSenderEthBalance);
```

Comment:
Actually, this should be done in one line. Bad practice to declare a var and overwrite it right away.
`const` variables catch errors at compile time, instead of run time.

Reply:
Thank you for all the ES6 tips. This is very valuable to me also.